File: solutions/csharp/lucians-luscious-lasagna/1/LuciansLusciousLasagna.cs
class Lasagna
{
    public int ExpectedMinutesInOven() => 40;
    
    public int RemainingMinutesInOven(int actualMinutesInOven)
        => ExpectedMinutesInOven() - actualMinutesInOven;
    
    public int PreparationTimeInMinutes(int numberOfLayers) => 2 * numberOfLayers;
    
    public int ElapsedTimeInMinutes(int numberOfLayers, int actualMinutesInOven)
        => PreparationTimeInMinutes(numberOfLayers) + actualMinutesInOven;
}

File: solutions/haskell/lucians-luscious-lasagna/1/src/LuciansLusciousLasagna.hs
module LuciansLusciousLasagna (elapsedTimeInMinutes, expectedMinutesInOven, preparationTimeInMinutes) where

expectedMinutesInOven :: Integer
expectedMinutesInOven = 40

preparationTimeInMinutes :: Num a => a -> a
preparationTimeInMinutes numOfLayers = numOfLayers * 2

elapsedTimeInMinutes :: Num a => a -> a -> a
elapsedTimeInMinutes numOfLayers actualMinutesInOven =
  preparationTimeInMinutes numOfLayers + actualMinutesInOven